src/communication/linear_spike_queue_policy.hpp

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#ifndef NMC_USE_LINEAR_SPIKE_QUEUE
+#error "linear_spike_queue_policy.hpp should only be compiled in a NMC_USE_LINEAR_SPIKE_QUEUE build"
+#endif
+
+#include <util/compare.hpp>
+#include "base_communicator.hpp"
+
+namespace nest {
+namespace mc {
+namespace communication {
+
+using nest::mc::util::make_range;
+
+template <typename CommunicationPolicy>
+class linear_spike_communicator: public base_communicator<CommunicationPolicy> {
+public:
+    using base = base_communicator<CommunicationPolicy>;
+    using typename base::event_queue;
+    using base::num_groups_local;
+    using base::base;
+    
+protected:
+    using base::cell_group_index;
+    using base::connections_;
+
+public:
+    // go over all spikes, search for connections with the same source as the spike
+    // and then push an event to each connection
+    // O(spikes * log (connections) * connections/spike)
+    std::vector<event_queue> make_event_queues(const gathered_vector<spike>& global_spikes)
+    {
+        auto queues = std::vector<event_queue>(num_groups_local());
+        
```

Comment:
Yes -- that's implemented in global_search. This one was done afterwards, to compare just the first spike and first connections effects, distinctly from the clustering effect (which is implemented).